sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruningSuite.scala

```diff
@@ -141,6 +141,14 @@ class PruningSuite extends HiveComparisonTest with BeforeAndAfter {
       Seq("2008-04-08", "11"),
       Seq("2008-04-09", "11")))
 
+  createPruningTest("Partition pruning - with nondeterministic fields",
+    "SELECT value, hr FROM srcpart1 WHERE ds = '2008-04-08' AND rand(7) < 1",
```

Comment:
Yes you are right. I thought the deterministic part can always be PPDed safely but it was not, in fact, the order of each part should also be considered. For example:
rand() < 0.01 AND partition_col = 'some_value'
should not be PPDed, but
partition_col = 'some_value' AND rand() < 0.01
still could be.
Thank you for your kindly reply!